draft-ietf-keytrans-protocol.md

```diff
@@ -1805,6 +1772,91 @@ each label are not provided, given that these can be cached from the
 original Search or Update query.
 
 
+# Third Parties
+
+## Management
+
+With the Third-Party Management deployment mode, a third party is responsible
+for the majority of the work of storing and operating the Transparency Log,
+while the Service Operator serves mainly to enforce access control and
+authenticate the addition of new entries. All user queries specified in
+{{user-operations}} are initially sent by users directly to the Service Operator
+to be forwarded to the Third-Party Manager if they pass access control.
+
+While other operations are forwarded by the Service Operator unchanged,
+`UpdateRequest` structures are forwarded to the Third-Party Manager with the
+Service Operator's signature attached:
+
+~~~ tls-presentation
+struct {
+  UpdateRequest request;
+  opaque signature<0..2^16-1>;
+} ManagerUpdateRequest;
+~~~
+
+The signature is computed as described in {{update-format}}. The Service
+Operator MUST maintain its own records (independent of the Third-Party Manager)
+for the greatest version of each label for the purpose of producing this
+signature.
+
+## Auditing
+
+With the Third-party Auditing deployment mode, the Service Operator obtains
+signatures from a lightweight Third-Party Auditor attesting to the fact that the
+Service Operator is constructing the tree correctly. These signatures are
+provided to users along with the responses to their queries.
+
+Each entry added to the log tree has a corresponding `AuditorUpdate` structure
+that allows the Third-Party Auditor to verify the log entry against stored state
+from previous log entries:
+
+~~~ tls-presentation
+struct {
+  uint64 timestamp;
+
+  PrefixLeaf added<0..2^16-1>;
+  PrefixLeaf removed<0..2^16-1>;
+
+  PrefixProof proof;
+} AuditorUpdate;
+~~~
```

Comment:
Assuming that these are served to auditors individually, and directly, it may be nice to allow logs to show diffs for more than one log entry, e.g., auditor last checked `n`, and log presents update for `n+m`. This should be easily doable by adding a `uint64 delta_to` field.

Reply:
I'm not sure what you mean here?

Reply:
This structure is basically a diff, right? The `timestamp` field encodes the newer version of the diff. My assumption was: Since the structure does not define a field for the older version, the diff is to the previous version (i.e., this structure will always represent a diff from epoch `n` to epoch `n+1`).

I think it would be better to allow for flexibility here and allow auditors to sign off diffs between epochs `n` and `n+m`. For that, we need two timestamps.

Reply:
AuditorUpdate structures aren't shown to users, they are sent from the service operator to the third-party auditor. The auditor's signature comes back as part of TreeHead, section 9.3. I will push some text that makes this clearer